src/BaseRelation.php

```diff
@@ -144,7 +144,7 @@ public function addEagerConstraints(array $models)
         // The first model in the array is always the parent, so add the scope constraints based on that model.
         // @link https://github.com/laravel/framework/pull/25240
         // @link https://github.com/lazychaser/laravel-nestedset/issues/351
-        optional($models[0])->applyNestedSetScope($this->query);
+        optional(reset($models))->applyNestedSetScope($this->query);
```

Comment:
https://github.com/lazychaser/laravel-nestedset/pull/552
https://github.com/lazychaser/laravel-nestedset/pull/547

Reply:
Ah, I see. The change from `$models[0]` to `reset($models)` was a fix which for some reason was only applied to their v5 (in lazychaser#547) but did not yet make it into their v6 (because lazychaser#552). So it is basically a upstream regression bug which you already fixed for "our" v6.